python/GafferSceneTest/EditScopeAlgoTest.py

```diff
@@ -1037,5 +1045,187 @@ def testProcessorNames( self ) :
+			Gaffer.MetadataAlgo.setReadOnly( component, True )
+			self.assertEqual(
+				GafferScene.EditScopeAlgo.optionEditReadOnlyReason( s["box"]["scope"], "test" ),
+				s["box"]
+			)
+
+		Gaffer.MetadataAlgo.setReadOnly( s["box"], False )
+		self.assertEqual(
+			GafferScene.EditScopeAlgo.optionEditReadOnlyReason( s["box"]["scope"], "test" ),
+			s["box"]["scope"]
+		)
+
+		Gaffer.MetadataAlgo.setReadOnly( s["box"]["scope"], False )
+		GafferScene.EditScopeAlgo.acquireOptionEdit( s["box"]["scope"], "test" )
+
+		self.assertIsNone(
+			GafferScene.EditScopeAlgo.optionEditReadOnlyReason( s["box"]["scope"], "test" )
+		)
+
+		candidateComponents = (
+			s["box"]["scope"]["OptionEdits"]["edits"][0]["value"],
+			s["box"]["scope"]["OptionEdits"]["edits"][0]["mode"],
+			s["box"]["scope"]["OptionEdits"]["edits"][0]["enabled"],
+			s["box"]["scope"]["OptionEdits"]["edits"][0]["name"],
+			s["box"]["scope"]["OptionEdits"]["edits"][0],
+			s["box"]["scope"]["OptionEdits"]["edits"],
+			s["box"]["scope"]["OptionEdits"],
+			s["box"]["scope"],
+			s["box"]
+		)
+
+		for component in candidateComponents :
+			Gaffer.MetadataAlgo.setReadOnly( component, True )
+			self.assertEqual(
+				GafferScene.EditScopeAlgo.optionEditReadOnlyReason(
+					s["box"]["scope"],
+					"test"
+				),
+				component
+			)
+
+		for component in candidateComponents :
+			Gaffer.MetadataAlgo.setReadOnly( component, False )
+
+		GafferScene.EditScopeAlgo.removeOptionEdit( s["box"]["scope"], "test" )
+
+		self.assertIsNone(
+			GafferScene.EditScopeAlgo.acquireOptionEdit(
+				s["box"]["scope"],
+				"test",
+				createIfNecessary = False
+			)
+		)
+
+		for component in (
+			s["box"]["scope"]["OptionEdits"]["edits"],
+			s["box"]["scope"]["OptionEdits"]
+		) :
+			Gaffer.MetadataAlgo.setReadOnly( component, True )
+			self.assertEqual(
+				GafferScene.EditScopeAlgo.optionEditReadOnlyReason(
+					s["box"]["scope"],
+					"test"
+				),
+				component
+			)
+
+	def testOptionEditRemoval( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		s["options"] = GafferScene.CustomOptions()
+		s["options"]["options"].addChild( Gaffer.NameValuePlug( "test", IECore.IntData( 10 ) ) )
+
+		s["scope"] = Gaffer.EditScope()
+		s["scope"].setup( s["options"]["out"] )
+		s["scope"]["in"].setInput( s["options"]["out"] )
+
+		edit1 = GafferScene.EditScopeAlgo.acquireOptionEdit( s["scope"], "test" )
+
+		self.assertEqual( len( s["scope"]["OptionEdits"]["edits"].children() ), 1 )
+		self.assertEqual( s["scope"]["OptionEdits"]["edits"][0], edit1 )
+
+		GafferScene.EditScopeAlgo.removeOptionEdit( s["scope"], "test" )
+
+		self.assertEqual( len( s["scope"]["OptionEdits"]["edits"].children() ), 0 )
+
+	def testOptionEditExceptions( self ) :
+
+		options = GafferScene.Options()
+		editScope = Gaffer.EditScope()
+		editScope.setup( options["out"] )
+		editScope["in"].setInput( options["out"] )
+		emptyKeys = editScope.keys()
+
+		with six.assertRaisesRegex( self, RuntimeError, 'Option "bogus" does not exist' ) :
```

Comment:
Note to self : various Python 2 things are getting commited to `1.1_maintenance` and will need removing on `main` once merged through.

Reply:
No need to fix it when merging, rebasing onto `1.2` did the job. Fixed in https://github.com/GafferHQ/gaffer/pull/4907/commits/47492f95d21f0afdd34801492f40e601cf7cbc5e